add a new recipe for ifcopenshell

EstebanDugueperoux2 opened the pull request:
### Summary
Add a date based version of IfcOpenShell (https://github.com/IfcOpenShell/IfcOpenShell)

#### Motivation
I want to add a such recipe to conan-center-index to ease IfcOpenShell C++ API consumption.
We study also the use of conan in CI of IfcOpenShell.

#### Details
This recipe build a main parser lib to be able to parse IFC files for differents possibles schemas.
In addition it provides geometry libraries for these IFC files geometry manipulation.

---
- [X] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [X] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [X] If this is a bug fix, please link related issue or provide bug details
- [X] Tested locally with at least one configuration using a recent version of Conan

---
Add a :+1: reaction to pull requests you find [important](https://github.com/conan-io/conan-center-index/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc) to help the team prioritize, thanks!

EstebanDugueperoux2 committed:
add a new recipe for ifcopenshell

franramirez688 commented:
Hi @EstebanDugueperoux2, thanks for this PR 🙏 

For the record, the CI is no longer running automatically, so it's up to the maintainers to run it for all the PRs.
When a PR is ready to be reviewed, we value it and run the CI.

This PR was a DRAFT, and it did not have any additional information, neither in the Changelog nor the code itself:
* Motivation to merge this PR in CCI 
* Details about it: upstream repo URL; is it a library? application? Any brief description?
* Comments on the code to help understand it faster:
   * Do they not release official versions? I saw the `cci.` versions.
   * Why is the patch file needed?
   * The test_package is quite large, so we should reduce it to a few lines.
   * So many options are declared; could we simplify them?

Thanks in advance!

EstebanDugueperoux2 commented:
Hi @franramirez688,

Thanks for the feedback.
I have updated the description of this PR.
IfcOpenShell team don't use a standard way to make release/tag, they do almost every day tag for unstable build.
The patch is mainly used to be able to build IfcOpenShell againsat cgal dependency in conan context.
I have simplified options to minimal.
I have also simplified test_package.cpp.

Is it ok for you?

Regards.

> Hi @EstebanDugueperoux2, thanks for this PR 🙏
> 
> For the record, the CI is no longer running automatically, so it's up to the maintainers to run it for all the PRs. When a PR is ready to be reviewed, we value it and run the CI.
> 
> This PR was a DRAFT, and it did not have any additional information, neither in the Changelog nor the code itself:
> 
> * Motivation to merge this PR in CCI
> * Details about it: upstream repo URL; is it a library? application? Any brief description?
> * Comments on the code to help understand it faster:
>   
>   * Do they not release official versions? I saw the `cci.` versions.
>   * Why is the patch file needed?
>   * The test_package is quite large, so we should reduce it to a few lines.
>   * So many options are declared; could we simplify them?
> 
> Thanks in advance!

EstebanDugueperoux2 committed:
Fix linter warnings

EstebanDugueperoux2 commented:
Hi @franramirez688 ,

Is it normal to have CI so long?

Regards.

franramirez688 committed:
Apply suggestion from @franramirez688

EstebanDugueperoux2 committed:
ifcopenshell: Try to fix MacOSX build and try to make rocksdb integration works
